app/javascript/controllers/chart_controller.js

```diff
@@ -141,6 +112,30 @@ export default class extends Controller {
     ]
   }
 
+  modifyDates() {
+    let dateOption
+    switch(this.aggValue) {
+      case "day":
+        dateOption = { weekday: "short", year: "2-digit", month: "short", day: "2-digit" };
+        break;
+      case "week":
+        dateOption = { year: "numeric", month: "short", day: "2-digit" };
+        break;
+      case "month":
+        dateOption = { year: "numeric", month: "short" };
+        break;
+      case "year":
+        dateOption = { year: "numeric"};
+        break;
+    }
+    const datesModify = this.datesValue.map((e) => { 
+      let d = new Date(e);
+      d = d.toLocaleDateString("en-US", dateOption);
+      return d;
+    });
+    return datesModify;
```

Comment:
Here also, you can return the return value of the map function directly instead of needing to define `datesModify`:

```js
return this.datesValue.map(...);
```